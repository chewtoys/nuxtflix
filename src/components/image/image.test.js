import { shallowMount } from "@vue/test-utils"
import Image from "./image.vue"

describe("Image", () => {
  test("renders required props", () => {
    const wrapper = shallowMount(Image, {
      propsData: {
        src: "/image.jpg",
        alt: "Foo"
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  test("renders optional props", () => {
    const wrapper = shallowMount(Image, {
      propsData: {
        src: "/image.jpg",
        alt: "Foo",
        width: 200,
        focus: "faces",
        flag: "progressive",
        quality: 50
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
