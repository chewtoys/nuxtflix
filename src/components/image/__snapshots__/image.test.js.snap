// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render optional props 1`] = `
<picture
  class="image"
>
  <source
    srcset="/image.jpg?fm=webp&q=50&w=200&f=faces&fl=progressive"
    type="image/webp"
  />
  <source
    srcset="/image.jpg?fm=jpg&q=50&w=200&f=faces&fl=progressive"
    type="image/jpeg"
  />
   
  <img
    alt="Foo"
    src="/image.jpg?q=50&w=200&f=faces&fl=progressive"
  />
</picture>
`;

exports[`render required props 1`] = `
<picture
  class="image"
>
  <source
    srcset="/image.jpg?fm=webp"
    type="image/webp"
  />
  <source
    srcset="/image.jpg?fm=jpg"
    type="image/jpeg"
  />
   
  <img
    alt="Foo"
    src="/image.jpg"
  />
</picture>
`;
